### PR TITLE
interp: run "Confirm" tests on CI

### LINKED ISSUE #785: interp: run "Confirm" tests on CI
Our interpreter test suite is checked against the real bash (as well as other shells, like mksh and dash). However, since those shells had bugs in older versions, the tests are skipped if the version is too old (or too new, as sometimes changes in behavior happen!).

This results in test failures being merged to master, as CI doesn't run those tests. For example, see https://github.com/mvdan/sh/commit/4a4c1600341f6413719f5e8a50105b8d13a96d6d, which fixed one such skipped test case.

We should make CI run those. For instance, we could use https://hub.docker.com/_/bash via https://github.com/mvdan/dockexec, which allows us to pin the same version of Bash that the tests want. We could possibly do something similar for mksh or other shells, but Bash is by far the most important one.

If anyone wants to contribute a patch: note that you can test this on GitHub Actions CI by forking the repo and pushing to a branch. You could also test how to use dockexec and that docker image with `go test` locally.

### PR DESCRIPTION
**Pull request by dhanusaputra**
update bash directly instead of use dockexec since got `no such file` error
close #785

### CODE REVIEW

**Comment by mvdan** [comment]
> update bash directly instead of use dockexec since got `no such file` error

Did you see https://github.com/mvdan/dockexec#caveats? I still think using dockexec would be better long-term; mixing the use of docker images is more composable than directly installing programs on the host system.

**Comment by dhanusaputra** [comment]
yes already tried that part but no luck
will try again

**Comment by mvdan** [comment]
If you get stuck, you can push your changes here and I can help debug what's going on.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -28,6 +28,13 @@ jobs:
       run: diff <(echo -n) <(gofmt -d .)
       if: matrix.os == 'ubuntu-latest'
 
+    - name: Install dockexec
+      run: go install mvdan.cc/dockexec@latest
+      if: matrix.os == 'ubuntu-latest'
+    - name: Test interpreter confirm
+      run: CGO_ENABLED=0 go test -run TestRunnerRunConfirm -exec 'dockexec bash:5.1' ./interp/.
```
Comment (mvdan) [inline .github/workflows/test.yml:35]: I would instead have one step that runs two commands, calling it something like `Run confirm tests with Bash 5.1`.

Also, you can just use `./interp` as the final argument.

**Review by mvdan — APPROVED** [review]
SGTM! Just one last nit.

**Review by mvdan — APPROVED** [review]
Thanks! cc @riacataquian

**Pull request merged**